pkg/telemetry/telemetryservice.go

```diff
@@ -113,10 +115,11 @@ func NewTelemetryService(notifier webhook.QueuedNotifier, analytics AnalyticsSer
 }
 
 func (t *telemetryService) FlushStats() {
-	t.lock.Lock()
-	defer t.lock.Unlock()
+	t.lock.RLock()
+	workersShadow := t.workersShadow
+	t.lock.RUnlock()
 
-	for _, worker := range t.workers {
+	for _, worker := range workersShadow {
 		worker.Flush()
 	}
```

Comment:
Using a shadow to reduce lock scoping here. I vaguely remember doing this before and there were concerns about this happening a lot (i. e. shadowing). But, I am thinking it should be fine as it happens only when a participant joins or when a stats worker is reaped. I can revert the change if shadowing is a concern. But, at least we need to change this to `RLock` as flushing on potentially 100s of participants with write lock will hold lock for long. All of these run in separate goroutines. So, it is probably fine, but still feels like holding write lock here is not necessary.